j1939/Dm14Query.py

```diff
@@ -0,0 +1,290 @@
+from enum import Enum
+import queue
+import j1939
+
+
+class QueryState(Enum):
+    IDLE = 1
+    WAIT_FOR_SEED = 2
+    WAIT_FOR_DM16 = 3
+    WAIT_FOR_OPER_COMPLETE = 4
+
+
+class Command(Enum):
+    ERASE = 0
+    READ = 1
+    WRITE = 2
+    STATUS_REQUEST = 3
+    OPERATION_COMPLETED = 4
+    OPERATION_FAILED = 5
+    BOOT_LOAD = 6
+    EDCP_GENERATION = 7
+
+
+class Dm15Status(Enum):
+    PROCEED = 0
+    BUSY = 1
+    OPERATION_COMPLETE = 4
+    OPERATION_FAILED = 5
+
+
+class Dm14Query:
+    def __init__(self, ca: j1939.ControllerApplication) -> None:
+        """
+        performs memory access queries using DM14-DM18 messaging.  Presently only read queries are supported
```

Comment:
This is the stuff that already existed, right?  If so, both read and write queries are, or at least were, supported.

Reply:
Yeah this stuff existed already I broke it out into a new module just to reduce the size of the memory_access module. Read and write queries were already supported but responding to them wasn't